src/Application/Messenger/Handler/GiftAidResultHandler.php

```diff
@@ -28,8 +28,10 @@ public function __invoke(Messages\Donation $donationMessage): void
             $donationMessage->id,
         ));
 
+        $this->entityManager->beginTransaction();
```

Comment:
What do you think about replacing this (and similar) with
```php
$this->entityManager->wrapInTransaction(function () use ($donationMessage) {
```

I think makes it a bit easier to see at a glance where the transaction ends and impossible to forget to end the transaction. Sometimes the deep nesting can be awkward but I think OK in this case and usually.

Reply:
Thanks. Also I just noticed that Doctrine's implementation of wrapInTransaction does a bit more than begin and commit - if the closure throws it rolls back the transaction and closes the entity manager, both of which are good.